net/tstun/wrap.go

```diff
@@ -153,7 +153,7 @@ func Wrap(logf logger.Logf, tdev tun.Device) *Wrapper {
 		// a goroutine should not block when setting it, even with no listeners.
 		bufferConsumed: make(chan struct{}, 1),
 		closed:         make(chan struct{}),
-		outbound:       make(chan tunReadResult),
+		outbound:       make(chan tunReadResult, 1),
```

Comment:
This needs a comment saying that it can be unbuffered, but the buffer improves performance.

Did you try with other numbers, or is 1 all the protocol allows?

Reply:
In theory the protocol allows more than 1, because we occasionally inject packets, but in practice, 1 is all that will ever matter.